### ISSUE TITLE
RectLabel Type causing error in TuriCreate

### ISSUE THREAD

**Issue by jaysumona2019** [ryouchinsa/Rectlabel-support#90]

**Comment by jaysumona2019** [comment]
1) I get the following error when I run TuriCreate, it does not like the word type:rectangle
Unexpected characters after last column. ""type":"rectangle""
Parse failed at token ending at: 
	'[{"label":"Human","type":"rectangle",[1;31m^[0m"coordinates":{"x":250,"y":487,"width":167,"height":336}}]'
2) if I do search and replace  ""type":"rectangle"" with null and run it, the text is not there but TuriCreate still complains as below:
Unexpected characters after last column 250,
[{“label":"Human","coordinates":{"x":250,[1;31m^[0m"y":487,"width":167,"height":336}}]

anyone ran into this issue?

**Comment by jaysumona2019** [comment]
hi I fixed the issue - it was an extra single quote supplied by RectLabel - I removed and it ran - took me 3 hrs to figure this out - I saw an Apple presentation and noticed the single quote is not there
ORIGINAL
'[{"label":"Human","type":"rectangle",[1;31m^[0m"coordinates":{"x":250,"y":487,"width":167,"height":336}}]'

SHOULD BE
[{"label":"Human","type":"rectangle",[1;31m^[0m"coordinates":{"x":250,"y":487,"width":167,"height":336}}]

**Comment by jaysumona2019** [comment]
sorry I meant below 
ORIGINAL
'[{"label":"Human","type":"rectangle","coordinates":{"x":250,"y":487,"width":167,"height":336}}]'
SHOULD BE
[{"label":"Human","type":"rectangle","coordinates":{"x":250,"y":487,"width":167,"height":336}}]

**Comment by ryouchinsa** [comment]
Thanks for figuring out the problem.

But we checked the csv line with an extra single quote in our environment and it works with Turi Create.
We mean step 2 and 3 works without any errors.
```
path,annotations
/Users/ryo/Desktop/test_annotations/test_data_mask/25024276-157392571702156-8268897881251905536-n.jpg,'[{"label":"Human","type":"rectangle","coordinates":{"x":250,"y":487,"width":167,"height":336}}]'
```

Are you using our python code to convert the CSV file to the sframe file?
If you were not using our python code to create sframe file, it would fail.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/convert_csv_to_sframe.py

Our python code reads the CSV file with the function and arguments below.
```
csv = pd.read_csv(csv_file, quotechar='\'', doublequote=True, converters={'annotations':JSONParser})
```

**Comment by ryouchinsa** [comment]
We updated so that RectLabel does not write the extra single quote.

```
path,annotations
/Users/ryo/Desktop/test_annotations/test_data_csv/1.jpg,[{"label":"person","type":"rectangle","coordinates":{"x":405,"y":463,"width":331,"height":909}},{"label":"bare feet","type":"rectangle","coordinates":{"x":393,"y":838,"width":62,"height":118}},{"label":"bare feet","type":"rectangle","coordinates":{"x":392,"y":881,"width":52,"height":102}}]
```

Now the exported CSV file could be read by Turi Create directly.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/train_turicreate.py

```
csv_file = sys.argv[1]
csv = tc.SFrame(csv_file)

image_path = csv['path'][0]
image_folder = str(os.path.split(image_path)[0])

data = tc.image_analysis.load_images(image_folder, recursive=False)
data = data.join(csv)

model = tc.object_detector.create(data, max_iterations=3)
model.save('mymodel.model')
```

The new update version 2.60 was released.
https://rectlabel.com/help#xml_to_csv_turicreate

**Comment by jaysumona2019** [comment]
Hello Thanks, I will try this out.  I also have another question for you.
How do I use RectLabel bounding box and images to develop  model using
Keras Tensor flow.  Do you have any documentation for that?

Thanks.

**Comment by ryouchinsa** [comment]
You can convert images and xml files to TFRecord file which can be used on Tensorflow Object Detection API.
https://rectlabel.com/help#tf_record

**Comment by jaysumona2019** [comment]
Hi I followed your guideline and have this question:  I understand
Rectlabel will generate TFRecord Files for my image as mentioned below.
But how do I generate map.pbtxt file.  From my experience on turicretae you
only need data and annotation file - so I am confused how to generate the
pbtxt file - does rectlabel generate that - can you help?

# From tensorflow/models/research/

gsutil cp pet_faces_train.record-* gs://${YOUR_GCS_BUCKET}/data/

gsutil cp pet_faces_val.record-* gs://${YOUR_GCS_BUCKET}/data/

gsutil cp object_detection/data/pet_label_map.pbtxt
gs://${YOUR_GCS_BUCKET}/data/pet_label_map.pbtxt

**Comment by ryouchinsa** [comment]
Press "Export label map for Tensorflow" to create the .pbtxt file.
https://rectlabel.com/help#label_map

**Comment by jaysumona2019** [comment]
sorry, I figured that out I already created pbtxt file using the export
label.  How do I convert the data into TFRecord format, I am looking into
some you tube video - it seems like they are converting xml file into csv
and then TFRecord.  Is Rectlabel already does that as it has both images
and xml files or I have to generate python script to convert?

**Comment by ryouchinsa** [comment]
Hi, just follow these instructions.
https://rectlabel.com/help#tf_record

Other users could convert to TFRecord file following those instructions.

We provide python script to convert some files into TFRecord file.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/rectlabel_create_pascal_tf_record.py

**Comment by jaysumona2019** [comment]
hi Thanks for your support, I followed your direction to run - but it did
not generate TFRecord file.  Is there an alternative way to generate
TFRecord file?  I get the below error:

python rectlabel_create_pascal_tf_record.py

Traceback (most recent call last):

  File "rectlabel_create_pascal_tf_record.py", line 146, in <module>

    tf.app.run()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/platform/app.py",
line 125, in run

    _sys.exit(main(argv))

  File "rectlabel_create_pascal_tf_record.py", line 128, in main

    image_files = dataset_util.read_examples_list(FLAGS.image_list_path)

  File
"/Applications/anaconda2/lib/python2.7/site-packages/object_detection-0.1-py2.7.egg/object_detection/utils/dataset_util.py",
line 59, in read_examples_list

    lines = fid.readlines()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/lib/io/file_io.py",
line 188, in readlines

    self._preread_check()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/lib/io/file_io.py",
line 85, in _preread_check

    compat.as_bytes(self.__name), 1024 * 512, status)

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/errors_impl.py",
line 528, in __exit__

    c_api.TF_GetCode(self.status.status))

tensorflow.python.framework.errors_impl.NotFoundError: train.txt; No such
file or directory

**Comment by ryouchinsa** [comment]
Looking at the error message, you need to press "Export train, val, and test.txt". Check on the "Full path" check box.
https://rectlabel.com/help#tf_record

`tensorflow.python.framework.errors_impl.NotFoundError: train.txt; No such
file or directory`

**Comment by jaysumona2019** [comment]
hi Ryo, I put the train.txt file in the directory but along with all the
other file images, xml file, pbtxt file - but it still gives me error.  Is
there another way to fix this issue?  Thanks for your help.

$ python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py

Traceback (most recent call last):

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 146, in <module>

    tf.app.run()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/platform/app.py",
line 125, in run

    _sys.exit(main(argv))

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 128, in main

    image_files = dataset_util.read_examples_list(FLAGS.image_list_path)

  File
"/Applications/anaconda2/lib/python2.7/site-packages/object_detection-0.1-py2.7.egg/object_detection/utils/dataset_util.py",
line 59, in read_examples_list

    lines = fid.readlines()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/lib/io/file_io.py",
line 188, in readlines

    self._preread_check()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/lib/io/file_io.py",
line 85, in _preread_check

    compat.as_bytes(self.__name), 1024 * 512, status)

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/errors_impl.py",
line 528, in __exit__

    c_api.TF_GetCode(self.status.status))

tensorflow.python.framework.errors_impl.NotFoundError: train.txt; No such
file or directory

**Comment by ryouchinsa** [comment]
Did you specify the file path to our python code?

This is an example which worked.

```
IMAGES_DIR=/Users/ryo/Desktop/test_annotations/support2
ANNOTATIONS_DIR=/Users/ryo/Desktop/test_annotations/support2/annotations
IMAGE_LIST_PATH=/Users/ryo/Desktop/test_annotations/support2/train.txt
LABEL_MAP_PATH=/Users/ryo/Desktop/test_annotations/support2/label_map.pbtxt
OUTPUT_PATH=/Users/ryo/Desktop/test_annotations/support2/train.record

python object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py --images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}" --image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}" --output_path="${OUTPUT_PATH}"
```

This user was in the same situation and could solve it.
https://github.com/ryouchinsa/Rectlabel-support/issues/89

**Comment by jaysumona2019** [comment]
hi Rio, Thanks for your help, with your update I am getting a different
error now.  I am attaching two file, the original file with the old error
and the new file with the new error listed below.  Can you take a look at
the file and see if anything is wrong.

python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 8

SyntaxError: Non-ASCII character '\xe2' in file
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py
on line 9, but no encoding declared; see
http://python.org/dev/peps/pep-0263/ for details

**Comment by ryouchinsa** [comment]
Looking at the error, comments in the python code are making the errors.
`on line 9, but no encoding declared; see`

We removed the comments from the python code.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/rectlabel_create_pascal_tf_record.py

Could you try with the updated python code?

**Comment by jaysumona2019** [comment]
if I just run your project as I is without making any change I get the
following error.  I think we need to update directory path to

$ python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 32

    with tf.gfile.GFile(image_file, 'rb') as fid:

       ^

IndentationError: expected an indented block


do we need to update the directory path to below line - can you give me an
example how to modify this line?


flags.DEFINE_string('images_dir', '', 'Full path to images directory.')


flags.DEFINE_string('annotations_dir', 'annotations', 'Relative path to
annotations directory from images_dir.')


flags.DEFINE_string('image_list_path', 'train.txt', 'Full path to image
list file which is one of the output files from "Export train, val, and
test.txt".')


flags.DEFINE_string('label_map_path', 'label_map.pbtxt', 'Full path to
label map file.')


flags.DEFINE_string('output_path', '', 'Full path to output TFRecord file.')


flags.DEFINE_boolean('include_masks', False, 'If you train Mask-RCNN, add
--include_masks otherwise you can remove it. Mask images are expected to be
png files and in the annotations folder.')

**Comment by ryouchinsa** [comment]
You do not have to edit the python code.

You have to edit the directory path and file path below.
Could you show us where are your images folder, annotations folder, train.txt, label_map.pbtxt, output folder to save the tfrecord file?

```
IMAGES_DIR=/Users/ryo/Desktop/test_annotations/support2
ANNOTATIONS_DIR=/Users/ryo/Desktop/test_annotations/support2/annotations
IMAGE_LIST_PATH=/Users/ryo/Desktop/test_annotations/support2/train.txt
LABEL_MAP_PATH=/Users/ryo/Desktop/test_annotations/support2/label_map.pbtxt
OUTPUT_PATH=/Users/ryo/Desktop/test_annotations/support2/train.record

python object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py --images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}" --image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}" --output_path="${OUTPUT_PATH}"
```

**Comment by jaysumona2019** [comment]
I put all of them into the Alldata folder - path to that is shown below -
main contents of all data folder attached here

/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata


can you please give me the updated python file with this path - so it
works..

thanks.

**Comment by ryouchinsa** [comment]
Again, this python code is working for other users.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/rectlabel_create_pascal_tf_record.py

Again, could you modify these directory and file paths for your environment?
```
IMAGES_DIR=/Users/ryo/Desktop/test_annotations/support2
ANNOTATIONS_DIR=/Users/ryo/Desktop/test_annotations/support2/annotations
IMAGE_LIST_PATH=/Users/ryo/Desktop/test_annotations/support2/train.txt
LABEL_MAP_PATH=/Users/ryo/Desktop/test_annotations/support2/label_map.pbtxt
OUTPUT_PATH=/Users/ryo/Desktop/test_annotations/support2/train.record

python object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py --images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}" --image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}" --output_path="${OUTPUT_PATH}"
```

If you could give us your sample data to us, we can check it works in our environment.

**Comment by jaysumona2019** [comment]
sorry I could not find the directory path you indicated above in the in the
python code you supplied.  The python code has flags.DEFINE_string - will
this be modified?

flags.DEFINE_string('annotations_dir', 'annotations', 'Relative path to
annotations directory from images_dir.’)


flags.DEFINE_string('image_list_path', 'train.txt', 'Full path to image
list file which is one of the output files from "Export train, val, and
test.txt”.')


flags.DEFINE_string('label_map_path', 'label_map.pbtxt', 'Full path to
label map file.’)


flags.DEFINE_string('output_path', '', 'Full path to output TFRecord file.’)


flags.DEFINE_boolean('include_masks', False, 'If you train Mask-RCNN, add
--include_masks otherwise you can remove it. Mask images are expected to be
png files and in the annotations folder.’)


flags.DEFINE_boolean('ignore_difficult_instances', False, 'Whether to
ignore difficult instances.')

**Comment by ryouchinsa** [comment]
You have to modify these directory and file paths for your environment.

```
IMAGES_DIR=/Users/ryo/Desktop/test_annotations/support2
ANNOTATIONS_DIR=/Users/ryo/Desktop/test_annotations/support2/annotations
IMAGE_LIST_PATH=/Users/ryo/Desktop/test_annotations/support2/train.txt
LABEL_MAP_PATH=/Users/ryo/Desktop/test_annotations/support2/label_map.pbtxt
OUTPUT_PATH=/Users/ryo/Desktop/test_annotations/support2/train.record
```

**Comment by ryouchinsa** [comment]
What happens if you copy and paste these lines onto your console and run?

```
IMAGES_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata
ANNOTATIONS_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/annotations
IMAGE_LIST_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.txt
LABEL_MAP_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/label_map.pbtxt
OUTPUT_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.record

python object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py --images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}" --image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}" --output_path="${OUTPUT_PATH}"
```

**Comment by jaysumona2019** [comment]
you mean they are not part of the python code, just run them from terminal ?

**Comment by ryouchinsa** [comment]
Yes. Could you copy and paste those lines onto your console and run?
Other users are doing the same process, too.

If it does not work, you have to change the directory and file pahts for your environment.

**Comment by jaysumona2019** [comment]
ok I ran as you suggested, I had to modify the python path to fit where I
store python file - it ran but a gave a new error in line 32


$ IMAGES_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata

$
ANNOTATIONS_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/annotations

$
IMAGE_LIST_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.txt

$
LABEL_MAP_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/label_map.pbtxt

$
OUTPUT_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.record

$

$ python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py
--images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}"
--image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}"
--output_path="${OUTPUT_PATH}"

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 32

    with tf.gfile.GFile(image_file, 'rb') as fid:

       ^

IndentationError: expected an indented block


I fixed the line 32 indent, but then it starts giving another error in line
56.  I attached they new python code that fixes line 32 error but creates
line 56 error - which I could not fix.


$ IMAGES_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata

$
ANNOTATIONS_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/annotations

$
IMAGE_LIST_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.txt

$
LABEL_MAP_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/label_map.pbtxt

$
OUTPUT_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.record

$

$ python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py
--images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}"
--image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}"
--output_path="${OUTPUT_PATH}"

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 56

    class_id = getClassId(obj['name'], label_map_dict)

    ^

IndentationError: unexpected indent

**Comment by ryouchinsa** [comment]
At first, It looks that your directory and file paths are working.
This is a great progress for us.

```
IMAGES_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata
ANNOTATIONS_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/annotations
IMAGE_LIST_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.txt
LABEL_MAP_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/label_map.pbtxt
OUTPUT_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.record
```

Then, for the indentation error, we have not experienced and been reported from our users yet.

As you can browse this python code on the web, our indentations are following normal rules.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/rectlabel_create_pascal_tf_record.py

> I fixed the line 32 indent
> with tf.gfile.GFile(image_file, 'rb') as fid:

In our code on the github page, the code is at line 37.
Are you using exactly the same code on the link?

What happens if you fix all indentation errors by yourself?

**Comment by jaysumona2019** [comment]
hi, Can you email me the python fie here? I am using your GitHub file in
the link you provided, the GitHub has no way to download the file, so I
copy and pasted the code in a text editor and renamed the file, may be
indentation got changed that way.  Thanks for your help.

**Comment by ryouchinsa** [comment]
You  can download pressing the Clone or download green button.

![github](https://user-images.githubusercontent.com/1954306/56004871-7aa0f080-5d08-11e9-9a06-3e1341822c7d.png)

[Rectlabel-support-master.zip](https://github.com/ryouchinsa/Rectlabel-support/files/3071168/Rectlabel-support-master.zip)

**Comment by jaysumona2019** [comment]
ok I downloaded your python code and ran - now I get a different error:  my
images are in jpg and png - do they need to be all changed to JPEG?

$ IMAGES_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata

$
ANNOTATIONS_DIR=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/annotations

$
IMAGE_LIST_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.txt

$
LABEL_MAP_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/label_map.pbtxt

$
OUTPUT_PATH=/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/train.record

$

$ python
/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py
--images_dir="${IMAGES_DIR}" --annotations_dir="${ANNOTATIONS_DIR}"
--image_list_path="${IMAGE_LIST_PATH}" --label_map_path="${LABEL_MAP_PATH}"
--output_path="${OUTPUT_PATH}"

(0,
'/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/363e56f0-00443512.jpg')

/Applications/anaconda2/lib/python2.7/site-packages/object_detection-0.1-py2.7.egg/object_detection/utils/dataset_util.py:75:
FutureWarning: The behavior of this method will change in future versions.
Use specific 'len(elem)' or 'elem is not None' test instead.

  if not xml:

(1,
'/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/0a0a0b1a-27d9fc44.jpg')

(2, '/Users/jay/Documents/IOSCVTRAININGTURICREATE/Alldata/person_339.png')

Traceback (most recent call last):

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 138, in <module>

    tf.app.run()

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/python/platform/app.py",
line 125, in run

    _sys.exit(main(argv))

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 132, in main

    tf_example = dict_to_tf_example(data, image_file, annotations_dir,
label_map_dict, FLAGS.include_masks, FLAGS.ignore_difficult_instances)

  File
"/Applications/anaconda2/lib/python2.7/site-packages/tensorflow/models/research/object_detection/dataset_tools/rectlabel_create_pascal_tf_record.py",
line 42, in dict_to_tf_example

    raise ValueError('Image format not JPEG')

ValueError: Image format not JPEG

**Comment by ryouchinsa** [comment]
Tensorflow document assumes input images are JPEG format.

You can use "Resize for all images and annotations" in Image menu.
https://rectlabel.com/help#resize_image

**Comment by jaysumona2019** [comment]
hi using your method, I am able to generate tfrecord file - just one note
that you can write in your instruction "convert all images to jpeg format
before starting the work as recallable can process all image type but
Tensorflow can not". thanks for your help
I have a issue in the very last step now - it is not related to Rectlabel,
I was wondering if you know any user already mentioned this or not?  I am
getting the error in creating dist/object_detection-0.1.tar.gz

$ python setup.py sdist

running sdist

running egg_info

writing requirements to object_detection.egg-info/requires.txt

writing object_detection.egg-info/PKG-INFO

writing top-level names to object_detection.egg-info/top_level.txt

writing dependency_links to object_detection.egg-info/dependency_links.txt

reading manifest file 'object_detection.egg-info/SOURCES.txt'

writing manifest file 'object_detection.egg-info/SOURCES.txt'

running check

warning: check: missing required meta-data: url


warning: check: missing meta-data: either (author and author_email) or
(maintainer and maintainer_email) must be supplied


copying files to object_detection-0.1...

copying object_detection.egg-info/PKG-INFO ->
object_detection-0.1/object_detection.egg-info

copying object_detection.egg-info/SOURCES.txt ->
object_detection-0.1/object_detection.egg-info

copying object_detection.egg-info/dependency_links.txt ->
object_detection-0.1/object_detection.egg-info

copying object_detection.egg-info/requires.txt ->
object_detection-0.1/object_detection.egg-info

copying object_detection.egg-info/top_level.txt ->
object_detection-0.1/object_detection.egg-info

Writing object_detection-0.1/setup.cfg

Creating tar archive

error: [Errno 13] Permission denied: 'dist/object_detection-0.1.tar.gz'

**Comment by ryouchinsa** [comment]
Thanks for your feedback.
We have not been reported the error yet.

When do you use this command to train on the Tensorflow Object Detection API?
`python setup.py sdist`

**Comment by jaysumona2019** [comment]
hi If you follow the document, it is the very last step in the GitHub you
supplied:
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_pets.md

# From tensorflow/models/research/

bash object_detection/dataset_tools/create_pycocotools_package.sh
/tmp/pycocotools

python setup.py sdist

**Comment by ryouchinsa** [comment]
If you use Google Cloud, go through until the end of the tutorial.
Otherwise just download the data and check the training script could work locally is enough.
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_locally.md

**Comment by jaysumona2019** [comment]
thank you , I will try this I have another question RectLabel for TuriCreate
In this section they are talking about def mask_to_bbox_coordinates(img):.
I used your tool to create bounding box, why do I need to use this method/
do you know?
https://apple.github.io/turicreate/docs/userguide/object_detection/data-preparation.html

**Comment by ryouchinsa** [comment]
You do not have to use the tutorial.
Because the sample annotation data is not the xml file but an png file.
So that they have to convert from png file to CSV data format.

After you export to CSV file, using the script below, you can start training.
https://github.com/ryouchinsa/Rectlabel-support/blob/master/train_turicreate.py
